File: Assignment2/app/src/main/java/ge/edu/freeuni/assignment2/ui/viewpager/ViewPagerAdapter.java
package ge.edu.freeuni.assignment2.ui.viewpager;

import java.util.ArrayList;
import java.util.List;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import ge.edu.freeuni.assignment2.model.location.Location;

public class ViewPagerAdapter extends FragmentStatePagerAdapter {
    private List<Location> data = new ArrayList<>();

    public ViewPagerAdapter(@NonNull FragmentManager fm) {
        super(fm);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        return WeatherFragment.newInstance(data.get(position).getName());
    }

    @Override
    public int getCount() {
        return data.size();
    }

    public void setData(List<Location> data) {
        this.data.clear();
        this.data.addAll(data);
        notifyDataSetChanged();
    }
}

File: 2/app/src/main/java/ge/edu/freeuni/rsr/tournaments/create/viewpager/ViewPagerAdapter.kt
package ge.edu.freeuni.rsr.tournaments.create.viewpager

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter


/*
* created by tgeldiashvili on 7/5/2019
*/

class ViewPagerAdapter(fm: FragmentManager) : FragmentStatePagerAdapter(fm, BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {

    companion object {
        const val FRAGMENT_CONFIG = 0
        const val FRAGMENT_QUESTION = 1
    }

    override fun getItem(position: Int): Fragment {
        return if (position == FRAGMENT_CONFIG)
            TournamentConfigFragment.newInstance()
        else
            QuestionInputFragment.newInstance()
    }

    override fun getCount(): Int {
        return 2
    }
}